common/src/autogluon/common/utils/utils.py

```diff
@@ -47,3 +50,62 @@ def setup_outputdir(path, warn_if_exist=True, create_dir=True, path_suffix=None)
     if path[-1] != os.path.sep:
         path = path + os.path.sep
     return path
+
+
+def get_python_version(include_micro=True) -> str:
+    if include_micro:
+        return f"{sys.version_info.major}.{sys.version_info.minor}.{sys.version_info.micro}"
+    else:
+        return f"{sys.version_info.major}.{sys.version_info.minor}"
+
+
+def get_package_versions() -> Dict[str, str]:
+    """Gets a dictionary of package name -> package version for every package installed in the environment"""
+    import pkg_resources
+    package_dict = pkg_resources.working_set.by_key
+    package_version_dict = {key: val.version for key, val in package_dict.items()}
+    return package_version_dict
+
+
+def get_autogluon_metadata() -> Dict[str, Any]:
+    from ..version import __version__
+    metadata = dict(
+        system=platform.system(),
+        version=f'{__version__}',
+        py_version=get_python_version(include_micro=False),
+        py_version_micro=get_python_version(include_micro=True),
+        packages=get_package_versions(),
+    )
+    return metadata
+
+
+def compare_autogluon_metadata(*, original: dict, current: dict, check_packages=True) -> list:
+    logs = []
+    og = original
+    cu = current
+    if og['version'] != cu['version']:
+        logs.append((30, f"WARNING: AutoGluon version mismatch (original={og['version']}, current={cu['version']})"))
+    if og['py_version'] != cu['py_version']:
+        logs.append((30, f"WARNING: AutoGluon Python version mismatch (original={og['py_version']}, current={cu['py_version']})"))
+    elif og['py_version_micro'] != cu['py_version_micro']:
+        logs.append((30, f"INFO: AutoGluon Python micro version mismatch (original={og['py_version_micro']}, current={cu['py_version_micro']})"))
```

Comment:
I think the warning have sense for 3.7 vs 3.8 (specifically because of serialization issues). Not sure if micro version should be a warning. Do we have any example of incompatibility for these?

Reply:
I don't know, currently it is just an info log and not a warning.

Reply:
If you want I can remove it so it is never logged